ray-curator/ray_curator/stages/text/io/reader/base.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base tabular reader stage for JSONL and Parquet readers."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+import pandas as pd
+import ray
+from loguru import logger
+
+if TYPE_CHECKING:
+    from ray_curator.backends.base import WorkerMetadata
+
+from ray_curator.backends.experimental.utils import RayStageSpecKeys
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.tasks import DocumentBatch, FileGroupTask
+
+
+@dataclass
+class BaseReader(ProcessingStage[FileGroupTask, DocumentBatch]):
+    """Common base for tabular file readers.
+
+    Subclasses must implement the read_data method.
+    """
+
+    fields: list[str] | None = None
+    read_kwargs: dict[str, Any] = field(default_factory=dict)
+    file_extensions: list[str] = field(default_factory=list)
+    _name: str = ""
+    _generate_ids: bool = False
+    _assign_ids: bool = False
+
+    def __post_init__(self) -> None:
+        if self._generate_ids and self._assign_ids:
+            msg = "Cannot generate and assign IDs at the same time"
+            raise ValueError(msg)
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return [], []
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], self.fields or []
+
+    def setup(self, _: WorkerMetadata | None = None) -> None:
+        if self._generate_ids or self._assign_ids:
+            from ray_curator.stages.deduplication.id_generator import get_id_generator_actor
```

Comment:
Can this be a top-level import?

Reply:
I think deduplication ends up importing GPU imports, so i kept is as is 
Earlier discussion here: 
- https://github.com/NVIDIA-NeMo/Curator/pull/907#discussion_r2274735023